### PR TITLE
8336039: Doccheck: HTML warnings, broken links and missing files in java.base documentation

### PR DESCRIPTION
**Pull request by nizarbenalla**
Can I get a review for this change that fixes some broken links in javadoc comments? The new docs are hosted [here](https://cr.openjdk.org/~nbenalla/GeneratedDocs/8336039-warnings-links/).

It's mostly fixing some relative links.
If using `{@docroot}` isn't ideal I can change it.

Here is the result of running `diff -r docs-master docs` on the docs from master vs and after these changes 

```
diff -r docs-master/api/java.base/java/lang/classfile/components/CodeStackTracker.html docs/api/java.base/java/lang/classfile/components/CodeStackTracker.html
106c106
<  <p>
---
>  </p>
diff -r docs-master/api/java.base/java/lang/classfile/package-summary.html docs/api/java.base/java/lang/classfile/package-summary.html
99c99
<  <p>
---
> 
106c106
<  <p>
---
> 
618c618
<  <p>
---
> 
755c755
<  <p>
---
> 
783c783
< <p>
---
> 
diff -r docs-master/api/java.base/java/lang/foreign/Arena.html docs/api/java.base/java/lang/foreign/Arena.html
142c142
<  the segments allocated by it) becomes <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>,
---
>  the segments allocated by it) becomes <a href="../../../java/lang/ref/package-summary.html#reachability">unreachable</a>,
diff -r docs-master/api/java.base/java/lang/foreign/MemorySegment.Scope.html docs/api/java.base/java/lang/foreign/MemorySegment.Scope.html
120c120
<  as long as it is <a href="../../../java/lang/ref/package.html#reachability">reachable</a>.
---
>  as long as it is <a href="../../../java/lang/ref/package-summary.html#reachability">reachable</a>.
diff -r docs-master/api/java.base/java/lang/foreign/MemorySegment.html docs/api/java.base/java/lang/foreign/MemorySegment.html
1420c1420
<           kept <a href="../../../java/lang/ref/package.html#reachability">reachable</a>
---
>           kept <a href="../../../java/lang/ref/package-summary.html#reachability">reachable</a>
1833c1833
<           <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>.
---
>           <a href="../../../java/lang/ref/package-summary.html#reachability">unreachable</a>.
1899c1899
<           <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>.
---
>           <a href="../../../java/lang/ref/package-summary.html#reachability">unreachable</a>.
diff -r docs-master/api/java.base/java/lang/foreign/SymbolLookup.html docs/api/java.base/java/lang/foreign/SymbolLookup.html
395c395
<  <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>. The
---
>  <a href="../../../java/lang/ref/package-summary.html#reachability">unreachable</a>. The
diff -r docs-master/api/java.base/java/net/spi/InetAddressResolver.LookupPolicy.html docs/api/java.base/java/net/spi/InetAddressResolver.LookupPolicy.html
117c117
<  <a href="doc-files/net-properties.html#Ipv4IPv6">System Properties</a> which affect
---
>  <a href="../doc-files/net-properties.html#Ipv4IPv6">System Properties</a> which affect
diff -r docs-master/api/java.base/java/text/MessageFormat.html docs/api/java.base/java/text/MessageFormat.html
433,434c433
< 
<  <p>
---
>  </p>
diff -r docs-master/api/java.base/java/util/concurrent/StructuredTaskScope.ShutdownOnFailure.html docs/api/java.base/java/util/concurrent/StructuredTaskScope.ShutdownOnFailure.html
245c245
<  <a href="#TreeStructure">Tree Structure</a> section in the class description
---
>  <a href="../../../../java.base/java/util/concurrent/StructuredTaskScope.html#TreeStructure">Tree Structure</a> section in the class description
diff -r docs-master/api/java.base/java/util/concurrent/StructuredTaskScope.ShutdownOnSuccess.html docs/api/java.base/java/util/concurrent/StructuredTaskScope.ShutdownOnSuccess.html
242c242
<  <a href="#TreeStructure">Tree Structure</a> section in the class description
---
>  <a href="../../../../java.base/java/util/concurrent/StructuredTaskScope.html#TreeStructure">Tree Structure</a> section in the class description
diff -r docs-master/api/java.base/javax/security/auth/Subject.html docs/api/java.base/javax/security/auth/Subject.html
207,208c207,208
<  whether a security manager is
<  <a href="../../../java/lang/SecurityManager.html#set-security-manager">allowed or disallowed</a></a></b>:
---
>  whether a security manager is</a>
>  <a href="../../../java/lang/SecurityManager.html#set-security-manager">allowed or disallowed</a></b>:
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issue
 * [JDK-8336039](https://bugs.openjdk.org/browse/JDK-8336039): Doccheck: HTML warnings, broken links and missing files in java.base documentation (**Bug** - P4)

### Reviewers
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20251/head:pull/20251` \
`$ git checkout pull/20251`

Update a local copy of the PR: \
`$ git checkout pull/20251` \
`$ git pull https://git.openjdk.org/jdk.git pull/20251/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20251`

View PR using the GUI difftool: \
`$ git pr show -t 20251`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20251.diff">https://git.openjdk.org/jdk/pull/20251.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20251#issuecomment-2238971913)

### CODE REVIEW

File: src/java.base/share/classes/javax/security/auth/Subject.java [hunk 1]
```diff
@@ -109,8 +109,8 @@
  * input type and exceptions thrown are slightly different.
  *
  * <p><b><a id="sm-allowed">These methods behave differently depending on
```
Comment (djelinski) [inline src/java.base/share/classes/javax/security/auth/Subject.java:111]:
```suggestion
 * <p id="sm-allowed"><b>These methods behave differently depending on
```
We seem to mix `<p id` and `<p><a id` in our code, so both forms should work fine, and the `<p id` form seems to be a better fit here.

File: src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java [hunk 2]
```diff
@@ -1010,7 +1010,7 @@ public static final class ShutdownOnSuccess<T> extends StructuredTaskScope<T> {
          *
          * <p> Construction captures the current thread's {@linkplain ScopedValue scoped
          * value} bindings for inheritance by threads started in the task scope. The
-         * <a href="#TreeStructure">Tree Structure</a> section in the class description
+         * <a href="{@docRoot}/java.base/java/util/concurrent/StructuredTaskScope.html#TreeStructure">Tree Structure</a> section in the class description
```
[the remark this answers is not in the document]
Comment (nizarbenalla) [reply src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java]: Fixed in [31649c7](https://github.com/openjdk/jdk/pull/20251/commits/31649c743e709cd67bd132774636e3484939890c), thanks.

File: src/java.base/share/classes/javax/security/auth/Subject.java (hunk 1 above)
Comment (nizarbenalla) [reply src/java.base/share/classes/javax/security/auth/Subject.java]: Fixed it, thanks. (I tried not to touch code beyond what was necessary for this PR)

File: src/java.base/share/classes/java/lang/classfile/components/CodeStackTracker.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
  * and calculating max stack size.
  * <p>
  * Sample use:
- * <p>
```
Comment (liach) [inline src/java.base/share/classes/java/lang/classfile/components/CodeStackTracker.java:40]: This `<p>` intends to add a break. So I think you can just remove this new paragraph, like you've done below.

File: src/java.base/share/classes/java/lang/foreign/Arena.java [hunk 4]
```diff
@@ -61,7 +61,7 @@
  * by the garbage collector. The scope of an automatic arena is an automatic scope. As
  * such, the regions of memory backing memory segments allocated with the automatic arena
  * are deallocated at some unspecified time <em>after</em> the automatic arena (and all
- * the segments allocated by it) becomes <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>,
+ * the segments allocated by it) becomes <a href="../../../java/lang/ref/package-summary.html#reachability">unreachable</a>,
```
Comment (liach) [inline src/java.base/share/classes/java/lang/foreign/Arena.java:64]:
```suggestion
 * the segments allocated by it) becomes {@linkplain java.lang.ref##reachability unreachable},
```
same for other occurrences.

File: src/java.base/share/classes/java/text/MessageFormat.java [hunk 5]
```diff
@@ -371,8 +371,7 @@
  * The following example demonstrates a general usage of {@code MessageFormat}.
  * In internationalized programs, the message format pattern and other
  * static strings will likely be obtained from resource bundles.
- *
- * <p>
+ * </p>
```
Comment (liach) [inline src/java.base/share/classes/java/text/MessageFormat.java:374]: Same here, just remove `<p>` instead of adding a closing

**Review by liach — APPROVED** [review]
Lgtm

/reviewers 2 reviewer

**Review by djelinski — APPROVED** [review]

File: src/java.base/share/classes/java/lang/foreign/Arena.java (hunk 4 above)
Comment (djelinski) [reply src/java.base/share/classes/java/lang/foreign/Arena.java]: TIL

File: src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java [hunk 6]
```diff
@@ -1197,7 +1197,7 @@ public static final class ShutdownOnFailure extends StructuredTaskScope<Object>
          *
          * <p> Construction captures the current thread's {@linkplain ScopedValue scoped
          * value} bindings for inheritance by threads started in the task scope. The
-         * <a href="#TreeStructure">Tree Structure</a> section in the class description
+         * {@linkplain StructuredTaskScope##TreeStructure Tree Structure} section in the class description
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java:1200]: You can drop the changes to STS if you want as we have an update coming that replaces these paragraphs/links.

Comment (nizarbenalla) [reply src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java]: Done, thanks for the heads up!

**Review by liach — APPROVED** [review]

**Review by djelinski — APPROVED** [review]

**Comment by nizarbenalla** [comment]
thank you for the reviews, again.

/integrate

**Pull request closed without merging**